src/Chapter2.hs

```diff
@@ -336,7 +336,20 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList firstIndex lastIndex list
+  | firstIndex < 0 || lastIndex < 0 || lastIndex < firstIndex || null list = []
+  | otherwise = traverse (lastIndex - firstIndex + 1) (moveTo firstIndex list)        
+    where
+      moveTo :: Int -> [a] -> [a]
+      moveTo index list
+        | null list = []
+        | index == 0 = list
+        | otherwise = moveTo (index - 1) (drop 1 list)
+      traverse :: Int -> [a] -> [a]
+      traverse count list
+        | count == 0 || null list = []
+        | otherwise = head list : traverse (count - 1) (drop 1 list)
```

Comment:
Wow, interesting! You made it with your own functions!

Here you use `head l` and `drop 1 l`. If you would just pattern match on the list, you would not need to do two operations (and one of them is unsafe).

Also, you might want to look at `take` and `drop` standard functions. They could be really helpful and you can use them instead of writing custom ones 🙂